client/src/components/dataset/sidebars/ChangeDataType.jsx

```diff
@@ -0,0 +1,207 @@
+        type="text"
+        value={defaultValue}
+        onChange={(event) => onChange(event.target.value)}
+      />
+    </div>
+  );
+}
+
+DefaultValueInput.propTypes = {
+  defaultValue: PropTypes.string,
+  onChange: PropTypes.func.isRequired,
+};
+
+
+const errorOptions = [
+  {
+    value: 'empty-cell',
+    label: 'Leave empty',
+  },
+  {
+    value: 'default-value',
+    label: 'Pick a default value',
+  },
+  {
+    value: 'fail',
+    label: 'Let the transform fail',
+  },
+  {
+    value: 'delete-row',
+    label: 'Delete the row',
+  },
+];
+
+export default class ChangeDataType extends Component {
+
+  constructor() {
+    super();
+    this.state = {};
+    this.handleChangeErrorStrategy = this.handleChangeErrorStrategy.bind(this);
+    this.mergeArgs = this.mergeArgs.bind(this);
+  }
+
+
+  componentWillMount() {
+    const { column, newColumnType } = this.props;
+    this.setState({
+      op: 'core/change-datatype',
+      args: {
+        columnName: column.columnName,
+        newType: newColumnType,
+        defaultValue: null,
+        dateFormat: 'YYYY-MM-DD',
+      },
+      onError: 'default-value',
+      errorStrategy: 'empty-cell',
+    });
+  }
+
+
+  mergeArgs(args) {
+    this.setState({ args: Object.assign({}, this.state.args, args) });
+  }
+
+  handleChangeErrorStrategy(errorStrategy) {
+    switch (errorStrategy) {
+      case 'emptyCell': {
+        const args = Object.assign({}, this.state.args, { defaultValue: null });
+        this.setState({
+          args,
+          onError: 'default-value',
+          errorStrategy,
+        });
+        break;
+      }
+      case 'default-value':
+      case 'fail':
+      case 'delete-row': {
+        this.setState({
+          onError: errorStrategy,
+          errorStrategy,
+        });
+        break;
+      }
+      default:
+        throw new Error(`Unkown error strategy ${errorStrategy}`);
+    }
+  }
+
+  render() {
+    const { column, dataTypeOptions, onClose, onApply } = this.props;
+    const { newType } = this.state.args;
+    return (
+      <div
+        className="DataTableSidebar"
+        style={{
+          width: '300px',
+          height: 'calc(100vh - 4rem)',
+        }}
+      >
+        <SidebarHeader onClose={onClose}>
+          Change data type for {column.title}
+        </SidebarHeader>
+        <div className="inputs">
+          <div className="inputGroup">
+            <label htmlFor="dataTypeMenu">
+              Change data type to:
+            </label>
+            <DashSelect
+              name="dataTypeMenu"
+              value={newType}
+              options={dataTypeOptions}
+              onChange={type => {
+                if (type === 'date') {
+                  this.mergeArgs({ newType: 'date', dateFormat: 'YYYY-MM-DD' });
+                }
+                this.mergeArgs({ newType: type });
+              }}
+            />
+          </div>
+          {newType === 'date' && column.type === 'text' ?
```

Comment:
If the `column.type` is of type `number` this is a number->date transform. I will interpret the number as a timestamp so no date format needs to be specified. Perhaps we later add the ability to interpret the number as year, in which case you would here pick between timestamp or year. We haven't discussed this so that's why I don't do it.

TLDR; the available date format options (DD-MM-YYYY, MM-DD-YYYY, etc.) doesn't make sense when the column type in number